src/main/kotlin/com/bridgecrew/services/ResultsCacheService.kt

```diff
@@ -4,31 +4,47 @@ import com.bridgecrew.CheckovResult
 import com.bridgecrew.results.*
 import com.bridgecrew.utils.CheckovUtils
 import com.intellij.openapi.components.Service
-import kotlin.io.path.Path
 import com.intellij.openapi.project.Project
 
 @Service
 class ResultsCacheService(val project: Project) {
     var checkovResults: MutableList<BaseCheckovResult> = mutableListOf()
     private val checkovResultsComparator: Comparator<BaseCheckovResult> = compareBy({ it.filePath }, { it.resource }, {it.severity})
     private val baseDir: String = project.basePath!!
+    private var selectedCategory: Category? = null
 
     fun getAllCheckovResults(): List<BaseCheckovResult> {
         return this.checkovResults
     }
 
+    fun updateCategory(category: Category?){
+        this.selectedCategory = category
+    }
 
     fun getCheckovResultsFilteredBySeverityGroupedByPath(severitiesToFilterBy: List<Severity>?): Map<String, List<BaseCheckovResult>> {
+        var filteredResults = checkovResults
         if(severitiesToFilterBy != null) {
-            checkovResults.filter {baseCheckovResult ->
+            filteredResults = filteredResults.filter { baseCheckovResult ->
                 severitiesToFilterBy.all { includedSeverity ->
                     includedSeverity == baseCheckovResult.severity
                 }
-            }
+            }.toMutableList()
+        }
+        if(selectedCategory != null) {
+            filteredResults = filteredResults.filter { baseCheckovResult ->
+                selectedCategory == baseCheckovResult.category
+            }.toMutableList()
         }
 
-        return this.checkovResults.groupBy { it.filePath }
+        return filteredResults.groupBy { it.filePath.toString() }
     }
+
+    fun getResultsByCategory(category: Category?): List<BaseCheckovResult> {
+        return if(category == null) checkovResults else checkovResults.filter { baseCheckovResult ->
+            category == baseCheckovResult.category
```

Comment:
repeated with line 35 maybe you can use a generic function to serve both scenario

Reply:
it's similar but not the same since in this case we're using one filter instead of two. Meaning we could create one function but then have to pass in the existing filtered list, which will create another check if we need to run on the entire list (send null in this case).